featuretools/utils/entity_utils.py

```diff
@@ -12,7 +12,7 @@
 ks = import_or_none('databricks.koalas')
 
 
-def infer_variable_types(df, link_vars, variable_types, time_index, secondary_time_index):
+def infer_variable_types(df, link_vars, variable_types, time_index, secondary_time_index, random_state=0):
```

Comment:
instead of adding a `random_state` option, can we use two data columns that will have the desire distribution regardless of the random state used for sampling?

Reply:
We can make data columns that most likely yield the desired distribution, but the test would still be a bit random, so there's no guarantee that the sample would always yield the desired distribution.

Reply:
If we make a data column where 99% of the rows are one value and 1% a second value, any sample will report under 5% unique values.

Likewise if we make a data column where every row is unique, we're guaranteed the sample will be above 5% unique values

Reply:
Actually I see what you're saying, but the idea that we would only sample 2 rows if nunique is 2 is confusing me

Reply:
I haven't looked at this in detail, but wouldn't we want `n` to be `10000` if the dataframe contains more than 10000 rows, or just the length of the dataframe if we have less than 10000? Basically if we have less than 10000 rows, use all the data, otherwise sample it?

Reply:
That's the logic that gets used for Categorical inference. Would this be different for some reason?

Reply:
The current implementation uses the number of unique values as the sample size. 

```python
nunique = column.nunique(dropna=False))
sample = column.sample(n=min(10000, nunique)
percent_unique = len(sample.unique()) / sample.size
```

If there were only two unique values, the sample size would only be two and we couldn't get below 5%.

Reply:
I think we should use the same logic to get the sample size that we use earlier in this method

```python
min(10000, len(df[variable]))
```

Reply:
Isn't using the number of unique values as the sample size the root of this issue? For columns that have a large number of rows and a small number of unique values, using only a small number of rows for the sample will be inadequate for getting a good estimation of the percent unique values in the column.

Reply:
Yeah, makes sense. I'll remove the random state.